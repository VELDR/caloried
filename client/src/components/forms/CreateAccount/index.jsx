import PropTypes from 'prop-types';
import { useState } from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { AlternateEmail, DoneAll, Info, Key, Person } from '@mui/icons-material';

import { nextStep, setAccount } from '@pages/SignUp/actions';
import { register } from '@containers/Client/actions';
import { selectAccount, selectActivityLevel, selectGoal, selectMetrics } from '@pages/SignUp/selectors';

import FormFooter from '@components/forms/FormFooter';
import FormHeader from '@components/forms/FormHeader';
import FormInput from '@components/ui/FormInput';

import classes from './style.module.scss';

const CreateAccount = ({ goal, metrics, activityLevel, account, intl: { formatMessage } }) => {
  const dispatch = useDispatch();

  const [accountData, setAccountData] = useState({
    username: account.username || '',
    email: account.email || '',
    password: '',
    confirmPassword: '',
  });

  const handleInputChange = (e) => {
    setAccountData({ ...accountData, [e.target.name]: e.target.value });
  };
  const { sex, dob, height, weight } = metrics;
  const { username, email } = accountData;

  const handleFormSubmit = () => {
    const combinedData = {
      ...accountData,
      goal,
      sex,
      dob,
      height,
      weight,
      activityLevel,
    };

    dispatch(setAccount({ username, email }));
    dispatch(
      register(combinedData, () => {
        dispatch(nextStep());
      })
    );
  };

  return (
    <div className={classes.container}>
      <FormHeader
        title={formatMessage({ id: 'app_account_title' })}
        description={formatMessage({ id: 'app_account_description' })}
      />
      <div className={classes.form}>
        <FormInput
          label={formatMessage({ id: 'app_username' })}
          icon={<Person />}
          type="text"
          name="username"
          value={accountData.username}
          onChange={handleInputChange}
        />
        <FormInput
          label={formatMessage({ id: 'app_email' })}
          icon={<AlternateEmail />}
          type="email"
          name="email"
          value={accountData.email}
          onChange={handleInputChange}
        />
        <FormInput
          label={formatMessage({ id: 'app_password' })}
          icon={<Key />}
          type="password"
          name="password"
          onChange={handleInputChange}
        />
        <FormInput
          label={formatMessage({ id: 'app_confirm_password' })}
          icon={<DoneAll />}
          type="password"
          name="confirmPassword"
          onChange={handleInputChange}
        />
        <div className={classes.form__info}>
          <Info /> <FormattedMessage id="app_info_verify" />
        </div>
      </div>
      <FormFooter onContinue={handleFormSubmit} />
    </div>
  );
};
CreateAccount.propTypes = {
  goal: PropTypes.string,
  metrics: PropTypes.object,
  activityLevel: PropTypes.number,
  account: PropTypes.object,
  intl: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  goal: selectGoal,
  metrics: selectMetrics,
  activityLevel: selectActivityLevel,
  account: selectAccount,
});

export default injectIntl(connect(mapStateToProps)(CreateAccount));
