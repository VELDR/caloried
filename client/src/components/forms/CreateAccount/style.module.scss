@import '@styles/base';

.container {
  @include flex-column(0.75rem);
  height: 100%;
  @include apply-at-md-size {
    justify-content: space-between;
  }
  .form {
    @include flex-center-column(1rem);
    @include apply-at-md-size {
      @include flex-row(1rem);
      margin: 0 auto;
      flex-wrap: wrap;
      max-width: 90%;
    }
    &__info {
      @include flex-center(0.25rem);
      font-size: $font-size-tiny;
      width: 75%;
      padding: 0.25rem 0.5rem;
      box-sizing: border-box;

      svg {
        @include box(1rem);
      }
    }
  }
}
