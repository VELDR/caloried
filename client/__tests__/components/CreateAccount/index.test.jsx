import CreateAccount from '@components/forms/CreateAccount';
import { fireEvent, render } from '@utils/testHelper';

let wrapper;

beforeEach(() => {
  wrapper = render(<CreateAccount />);
});

describe('CreateAccount Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('create-account')).toBeInTheDocument();
    expect(getByTestId('form-header')).toBeInTheDocument();
    expect(getByTestId('form-footer')).toBeInTheDocument();
  });

  it('should handle input change', () => {
    const { getByTestId } = wrapper;

    const usernameInput = getByTestId('form-input-text');

    fireEvent.change(usernameInput, { target: { value: 'testuser' } });

    expect(usernameInput.value).toBe('testuser');
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
