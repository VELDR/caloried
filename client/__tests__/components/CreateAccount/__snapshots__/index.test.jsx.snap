// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="create-account"
      >
        <div
          class="header"
          data-testid="form-header"
        >
          <div
            class="header__title"
          >
            Buat Akun Anda
          </div>
          <div
            class="header__description"
          >
            Anda tinggal selangkah lagi! Buat akun Anda untuk memulai!
          </div>
        </div>
        <div
          class="form"
        >
          <div
            class="form"
          >
            <div
              class="form__label"
            >
              Nama user
            </div>
            <div
              class="form__input"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
              <input
                data-testid="form-input-text"
                name="username"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form"
          >
            <div
              class="form__label"
            >
              Email
            </div>
            <div
              class="form__input"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AlternateEmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                />
              </svg>
              <input
                data-testid="form-input-email"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="form"
          >
            <div
              class="form__label"
            >
              Kata Sandi
            </div>
            <div
              class="form__input"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 10h-8.35C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H13l2 2 2-2 2 2 4-4.04L21 10zM7 15c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                />
              </svg>
              <input
                data-testid="form-input-password"
                name="password"
                type="password"
                value=""
              />
              <div
                class="eyeIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="form"
          >
            <div
              class="form__label"
            >
              Konfirmasi Kata Sandi
            </div>
            <div
              class="form__input"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DoneAllIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 7-1.41-1.41-6.34 6.34 1.41 1.41L18 7zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12-1.42-1.41zM.41 13.41 6 19l1.41-1.41L1.83 12 .41 13.41z"
                />
              </svg>
              <input
                data-testid="form-input-password"
                name="confirmPassword"
                type="password"
                value=""
              />
              <div
                class="eyeIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="form__info"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
             
            OTP akan dikirim ke email Anda untuk mengaktifkan akun Anda
          </div>
        </div>
        <div
          class="footer"
          data-testid="form-footer"
        >
          <button
            class="button footer__right footer__button"
            data-testid="primary-button"
            type="submit"
          >
            Mulai
             
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EastIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="create-account"
    >
      <div
        class="header"
        data-testid="form-header"
      >
        <div
          class="header__title"
        >
          Buat Akun Anda
        </div>
        <div
          class="header__description"
        >
          Anda tinggal selangkah lagi! Buat akun Anda untuk memulai!
        </div>
      </div>
      <div
        class="form"
      >
        <div
          class="form"
        >
          <div
            class="form__label"
          >
            Nama user
          </div>
          <div
            class="form__input"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <input
              data-testid="form-input-text"
              name="username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="form"
        >
          <div
            class="form__label"
          >
            Email
          </div>
          <div
            class="form__input"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AlternateEmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
              />
            </svg>
            <input
              data-testid="form-input-email"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="form"
        >
          <div
            class="form__label"
          >
            Kata Sandi
          </div>
          <div
            class="form__input"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 10h-8.35C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H13l2 2 2-2 2 2 4-4.04L21 10zM7 15c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
              />
            </svg>
            <input
              data-testid="form-input-password"
              name="password"
              type="password"
              value=""
            />
            <div
              class="eyeIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="form"
        >
          <div
            class="form__label"
          >
            Konfirmasi Kata Sandi
          </div>
          <div
            class="form__input"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DoneAllIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m18 7-1.41-1.41-6.34 6.34 1.41 1.41L18 7zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12-1.42-1.41zM.41 13.41 6 19l1.41-1.41L1.83 12 .41 13.41z"
              />
            </svg>
            <input
              data-testid="form-input-password"
              name="confirmPassword"
              type="password"
              value=""
            />
            <div
              class="eyeIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="form__info"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
           
          OTP akan dikirim ke email Anda untuk mengaktifkan akun Anda
        </div>
      </div>
      <div
        class="footer"
        data-testid="form-footer"
      >
        <button
          class="button footer__right footer__button"
          data-testid="primary-button"
          type="submit"
        >
          Mulai
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EastIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
